js/ui/tile_view.js

```diff
@@ -218,7 +218,12 @@ const TileView = CollectionWidget.inherit({
 
         this._arrangeItems(items);
         if(windowUtils.hasWindow()) {
-            this._$container[config.mainDimension](this._cells.length * this.option(config.baseItemMainDimension) + (this._cells.length + 1) * itemMargin);
+            const actualContentSize = this._cells.length * this.option(config.baseItemMainDimension) + (this._cells.length + 1) * itemMargin;
+            const containerSize = this._$container[config.mainDimension]();
+
+            if(actualContentSize > containerSize) {
+                this._$container[config.mainDimension](actualContentSize);
+            }
```

Comment:
Let's extract this code as a separate method to reduce the `renderGeometry` method and make it clear